src/Workleap.OpenApi.MSBuild/ContractFirstProcess.cs

```diff
@@ -0,0 +1,87 @@
+﻿namespace Workleap.OpenApi.MSBuild;
+
+/// <summary>
+/// For a Contract First approach it will:
+///     1. Validate the OpenAPI specification files base on spectral rules
+///     2. If validateCodeSync=true, will generate the OpenAPI specification files from the code and validate if it match the provided specifications.
+/// </summary>
+internal class ContractFirstProcess
+{
+    private readonly ILoggerWrapper _loggerWrapper;
+    private readonly SpectralManager _spectralManager;
+    private readonly SwaggerManager _swaggerManager;
+    private readonly OasdiffManager _oasdiffManager;
+
+    internal ContractFirstProcess(ILoggerWrapper loggerWrapper, SpectralManager spectralManager, SwaggerManager swaggerManager, OasdiffManager oasdiffManager)
+    {
+        this._loggerWrapper = loggerWrapper;
+        this._spectralManager = spectralManager;
+        this._swaggerManager = swaggerManager;
+        this._oasdiffManager = oasdiffManager;
+    }
+
+    internal async Task<bool> Execute(
+        string[] openApiSpecificationFiles,
+        string openApiToolsDirectoryPath,
+        string[] openApiSwaggerDocumentNames,
+        string openApiSpectralRulesetUrl,
+        bool validateCodeSync,
```

Comment:
Not a big fan of booleans as parameters in general. I personally prefer enums.

I don't mind if you want to keep it as is.